benchmark_runner/benchmark_operator/benchmark_operator_workloads.py

```diff
@@ -797,6 +797,34 @@ def run_workload_func(self, workload_full_name: str):
         # remove workload yaml at the end of run
         self.__remove_run_workload_yaml_file(workload_full_name=workload_full_name)
 
+    @typechecked
+    @logger_time_stamp
+    def run_workload_func(self, workload_full_name: str):
+        extract_prometheus_snapshot = self.__environment_variables_dict.get('run_artifacts_path', 'True')
+        if extract_prometheus_snapshot.lower() == 'true':
+            log_path = self.__environment_variables_dict.get('run_artifacts_path', '')
+            if not os.path.isdir(log_path):
+                os.mkdir(log_path)
+            logger.info('Deleting prometheus-k8s-0 pod')
+            # TODO: this won't work for Kubernetes
+            self.__oc.terminate_pod_sync(pod_name='prometheus-k8s-0', namespace='openshift-monitoring')
+            logger.info('Waiting for prometheus-k8s-0 pod to reappear')
+            self.__oc.wait_for_pod_ready(pod_name='prometheus-k8s-0', namespace='openshift-monitoring')
+            start_timestamp = self.__oc.exec(pod_name='prometheus-k8s-0', namespace='openshift-monitoring', command=f"date '+%Y_%m_%dT%H_%M_%S%z'")
+            logger.info('Waiting 60 seconds for prometheus to stabilize')
+            time.sleep(60)
+            self.run_workload_func_internal(workload_full_name)
+            logger.info('Waiting 120 seconds for metrics collection to complete')
+            time.sleep(120)
```

Comment:
No need waiting notification every time. Put only one

Reply:
As I said I think you create a separate method for cpature_prometheus_snapshot(action=start/end) and not add inside 
run_workload_func
What do you think ?

Reply:
The notifications for now, for testing; I plan to remove it prior to merge.

I'm not too keen on separating out start/end here.  The two operations are quite different, and it's not a huge amount of code.